src/source/query_features.js

```diff
@@ -24,6 +24,7 @@ export function queryRenderedFeatures(sourceCache: SourceCache,
             wrappedTileID: tileIn.tileID.wrapped().key,
             queryResults: tileIn.tile.queryRenderedFeatures(
                 styleLayers,
+                sourceCache.getFeatureState.bind(sourceCache),
```

Comment:
Passing the function because it maintains different feature states for each `sourceLayer`.

Reply:
Hmm -- there are several other places in the code that already have awareness of `_geojsonTileLayer`...  as such, I'd be inclined to say we should just dupe the logic, or else move it into `SourceFeatureState`.

Reply:
@anandthakker What is the advantage of duplicating the logic over passing the bound function? This PR has already reduced the _spred_ of `_geojsonTileLayer` by consolidating it in `SourceCache.`

Reply:
It just seems clearer to me: I find a callback as a method of passing data to be harder to read and understand as compared to passing an object.  Would moving the `_geojsonTileLayer` logic into `SourceFeatureState` give us the best of both worlds?

Reply:
> Would moving the _geojsonTileLayer logic into SourceFeatureState give us the best of both worlds?

I don't think so. I could change the name of the callback from `getState` to `queryState`, but all of these changes seem minor and not that significant advantages to whats already here.

Reply:
I agree with @anandthakker that moving the logic into `SourceFeatureState` or duping the logic would be a bit clearer than passing the callback here